vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -949,8 +949,19 @@ def grouped_topk(
     num_expert_group: int = 0,
     topk_group: int = 0,
     scoring_func: str = "softmax",
-    e_score_correction_bias: Optional[torch.Tensor] = None
+    e_score_correction_bias: Optional[torch.Tensor] = None,
 ) -> tuple[torch.Tensor, torch.Tensor]:
+    if num_expert_group <= 32 and topk <= 32 and \
+            e_score_correction_bias is not None:
```

Comment:
What decides this - is it a heuristic or what is implemented?

Reply:
Thanks for your comment!

This is based on what is implemented, see `grouped_topk_kernels.cu`:

```
  TORCH_CHECK(n_group <= 32,
              "n_group should be smaller than or equal to 32 for now");
  TORCH_CHECK(topk <= 32, "topk should be smaller than or equal to 32 for now");
```